### DISCUSSION TITLE
Is it possible to update on-deployed trigger to only run when an app is updated

### DISCUSSION THREAD

**Discussion by smeeklai** [argoproj-labs/argocd-notifications#299] [Q&A]
The way we use ArgoCD is slightly different from how it should be. We have only one ArgoCD repo for an entire project. A project consists of several applications. The target revision of all the apps is `master`. Right now, there're 2 triggers, `on-deployed` and `on-sync-running`. Whenever a PR of an app is merged to `master`, all ArgoCd apps are triggered but only one will be updated as it has changed. However, since all ArgoCD apps are triggered, the `argocd-notifications` service's `on-deployed` trigger is executed and sent notifications because its condition is
> when: app.status.operationState.phase in ['Succeeded'] and app.status.health.status == 'Healthy'

So my question is, is it possible to update the trigger's condition to ignore the apps that don't get updated and only send the notification of the one that was actually get updated?

**Comment by ryota-sakamoto** [comment]
Hi @smeeklai 
I would like to know what it means `only one ArgoCD repo`.
Would you share more details about your configuration, please?

**Comment by wayttjl** [comment]
In my case,i used onceper like this:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  annotations:
    example.com/version: v0.1
```
oncePer: app.metadata.annotations["example.com/version"]

**Comment by Rohit-Tanwar** [comment]
@smeekli -- Is yous issue resolved? I also have the same scenario getting multiple email alerts when anything merged the main repo.
